addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -204,6 +204,18 @@ const Wysiwyg = Widget.extend({
                         $field.attr('contenteditable', false);
                         const $currencyValue = $field.find('.oe_currency_value');
                         $currencyValue.attr('contenteditable', true);
+                        $currencyValue.on('keydown', (e) => {
+                            // Valid key codes are :
+                            // 8: backspace/delete,
+                            // 37/39:left/right arrow,
+                            // 46: Delete
+                            const specialKeyCodes = [8, 37, 39, 46];
+                            const validKeyRegex = "[0-9\\.]";
+                            const keycode = e.which || e.keycode;
+                            if (!(e.key && e.key.match(validKeyRegex)) && !specialKeyCodes.includes(keycode)) {
+                                e.preventDefault();
+                            }
```

Comment:
I honestly do not like this method of validating the input. I feel constrained just by reading it. I'd be more in favor of a post validation. If you entered something that is not a valid price, you get a red border around the price and a little message that says "woops price invalid". That being said, this method has the advantage of preventing you from falling into those invalid cases to begin with, which is not bad I guess. However, because of its limiting nature, I'm not sure it's going to be that easy to make it work properly accross all languages. For example, this code won't work in french where the correct decimal separator is `,` rather than `.`. Technically you might even want to accept both as `.` is also valid in french as it is the thousand separator. So, `23.076,99` is a valid price in french, but not in english where it is equivalent to `23,076.99`. Actually, even if you only allow the decimal separator of the current user's language you can still end up with things like `1.9.3` which is not a valid price in any language. I think the only true way of solving this issue is to validate the price against the format of the current user's language.

Reply:
Thanks !

I didn't like this solution but wanted to give a try.
Your suggestion makes more sense and will lead to a solution which will be more maintainable.